### PR TITLE
[SceneKit] Make SCNMatrix4 column-major in .NET. Fixes #4652.

### LINKED ISSUE #4652: XAMCORE_4_0: Our managed SCNMatrix4 is row-major, while the native SCNMatrix4 is column-major.
Our managed SCNMatrix4 is row-major, while the native SCNMatrix4 is column-major.

This should be fixed, but it's a breaking change.

See #4525 for more information.

### PR DESCRIPTION
**Pull request by rolfbjarne**
* Implement a column-major version of SCNMatrix4 in .NET to match native code.
* This was done by copying the existing SCMatrix4 implementation, and modify it
  as required (doing it with conditional compilation in the same file turned out
  to be quite messy, so I opted for using different files for legacy Xamarin and
  .NET).
* There was one major change: the matrix inversion algorithm is new (copied from
   .NET instead), because the legacy Xamarin version showed strange results with
  some test values.
* Add setters for SCNMatrix4.Column[0-3] for legacy Xamarin to match the .NET API.
* Add CreateFromColumns methods for legacy Xamarin to match the .NET API.
* Add tests for all the new API.

Fixes https://github.com/xamarin/xamarin-macios/issues/4652.

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]

File: src/SceneKit/SCNMatrix4_dotnet.cs [hunk 1]
```diff
@@ -0,0 +1,1201 @@
...
+
+			pfloat x = (2.0f * zNear) / (right - left);
+			pfloat y = (2.0f * zNear) / (top - bottom);
+			pfloat a = (right + left) / (right - left);
+			pfloat b = (top + bottom) / (top - bottom);
+			pfloat c = -(zFar + zNear) / (zFar - zNear);
+			pfloat d = -(2.0f * zFar * zNear) / (zFar - zNear);
+
+			result = new SCNMatrix4 (x, 0, 0, 0,
+								 0, y, 0, 0,
```
Comment (tj-devel709) [inline src/SceneKit/SCNMatrix4_dotnet.cs:747]: Spacing looks maybe different than desired

File: src/SceneKit/SCNMatrix4_dotnet.cs [hunk 2]
```diff
@@ -0,0 +1,1201 @@
...
+		/// <param name="up">Up vector in world space (should not be parallel to the camera direction, that is target - eye)</param>
+		/// <returns>A SCNMatrix4 that transforms world space to camera space</returns>
+		public static SCNMatrix4 LookAt (SCNVector3 eye, SCNVector3 target, SCNVector3 up)
+		{
+			SCNVector3 z = SCNVector3.Normalize (eye - target);
+			SCNVector3 x = SCNVector3.Normalize (SCNVector3.Cross (up, z));
+			SCNVector3 y = SCNVector3.Normalize (SCNVector3.Cross (z, x));
+
+			SCNMatrix4 rot = new SCNMatrix4 (new SCNVector4 (x.X, y.X, z.X, 0.0f),
+										new SCNVector4 (x.Y, y.Y, z.Y, 0.0f),
```
Comment (tj-devel709) [inline src/SceneKit/SCNMatrix4_dotnet.cs:869]: Here as well

**Review by tj-devel709 — APPROVED** [review]
Some small possible spacing issues, but else good with me!

**Comment by rolfbjarne** [comment]
CC @mattleibow in case this affects MAUI

**Comment by rolfbjarne** [comment]
Test failures are unrelated:

* monotouch-test/Mac [dotnet]/Debug [dotnet]: https://github.com/xamarin/maccore/issues/2539
* monotouch-test/Mac [dotnet]/Debug (static registrar) [dotnet]: https://github.com/xamarin/maccore/issues/2539
* fsharp/Mac Catalyst [dotnet]/Debug [dotnet]: https://github.com/xamarin/maccore/issues/2414

**Pull request merged**